wkconnect/backends/tiff/models.py

```diff
@@ -74,20 +75,28 @@ def read_data(
         mag_factor = 2 ** zoom_step
         wk_offset_scaled = wk_offset // mag_factor
 
-        tile_coordinate_offset, tile_data = self.read_tile(
-            layer_name,
-            zoom_step,
-            (wk_offset_scaled.x, wk_offset_scaled.y),
-            (shape.x, shape.y),
-        )
-        left_in_tile = wk_offset_scaled[0] - tile_coordinate_offset[0]
-        top_in_tile = wk_offset_scaled[1] - tile_coordinate_offset[1]
-        right_in_tile = left_in_tile + shape.x
-        bottom_in_tile = top_in_tile + shape.y
-
-        cropout = tile_data[
-            top_in_tile:bottom_in_tile, left_in_tile:right_in_tile
-        ].transpose()
+        _, _, _, tile_shapes, _ = self.read_properties(layer_name)
+        if tile_shapes is None:
+            page_data = self.read_whole_page(layer_name, zoom_step)
+            cropout = page_data[
+                wk_offset_scaled[0] : wk_offset_scaled[0] + shape.x,
+                wk_offset_scaled[1] : wk_offset_scaled[1] + shape.y,
+            ]
+        else:
+            tile_coordinate_offset, tile_data = self.read_tile(
+                layer_name,
+                zoom_step,
+                (wk_offset_scaled.x, wk_offset_scaled.y),
+                (shape.x, shape.y),
+            )
+            left_in_tile = wk_offset_scaled[0] - tile_coordinate_offset[0]
+            top_in_tile = wk_offset_scaled[1] - tile_coordinate_offset[1]
+            right_in_tile = left_in_tile + shape.x
+            bottom_in_tile = top_in_tile + shape.y
+
+            cropout = tile_data[
+                top_in_tile:bottom_in_tile, left_in_tile:right_in_tile
+            ].transpose()
```

Comment:
I'm wondering why we need to transpose this once, and not for small tiffs. Mostly curiosity :cat2:

Reply:
for the small tifs I did the transpose already in the read_whole_page call. Since that is cached it only has to run once, which i preferred, but I don’t have a strong opinion on that, could also move it out to this calling place to unify it with the else branch. What do you think?

Reply:
I think that's perfectly fine as it is, just add a comment about it :+1: